### DISCUSSION TITLE
Unexpected colorspace/profile behaviour with layers

### DISCUSSION THREAD

**Discussion by axeld** [ImageMagick/ImageMagick#2575] [Help]
I am trying to convert an image like the following:
[bucket.zip](https://github.com/ImageMagick/ImageMagick/files/5222811/bucket.zip)

It is a CMYK TIFF file using Photoshop layers. If I just want to convert it to PNG, everything is fine, and works as expected:
`$ magick convert bucket.tif[0] bucket.png`

However, if I want to preserve the transparency, it's a bit more complicated, and results in wrong colors:
`$ magick convert -size 150x106 canvas:none bucket.tif[2] -background none -layers merge +repage bucket.png`

The only way I managed to preserve the colors is as follows:
`$ magick convert bucket.tif[0] bucket.icc`
`$ magick convert -size 150x106 canvas:none -colorspace CMYK -profile bucket.icc bucket.tif[2] -background none -layers merge +repage bucket.png`

Is there maybe a simpler way?

**Comment by fmw42** [comment]
Your image has a clip path that controls the transparency. So you need to use that. Also PNG does not support CMYK, so you should convert to sRGB using profiles. However, you clip path is not defined well and is off at the top.

Try this

`magick -quiet bucket.tif[0] -profile path_to/sRGB.icc -alpha transparent -clip -alpha opaque -channel a -negate +channel result.png
`

![result](https://user-images.githubusercontent.com/13458218/93174764-60302b00-f6e3-11ea-8b82-c5f24db31658.png)

Note that for IM 7, use magick, not magick convert.

**Reply by axeld** [reply]
Thanks, for looking into it! The original file is much larger, but you can also see in this version that the border looks bad, and that the handles are gone; I don't think the clipping path servers any useful purpose anymore.

I have to use layer 2 to get the result in good quality. I actually forgot to set the profile above, but it only improves the result if the original colors were correct:

`magick -quiet -size 150x106 canvas:none bucket.tif[2] -layers merge -profile  pathTo/sRGB.icc result.png`

![result](https://user-images.githubusercontent.com/863699/93179694-54218900-f736-11ea-88ff-000cf4b0a82e.png)

If I only set the colorspace to CMYK beforehand, the colors don't look good (even though the result looks already much closer to the desired one, best noticeable when looking at the red top cover):

`magick -quiet -size 150x106 -colorspace CMYK canvas:none bucket.tif[2] -layers merge -profile  pathTo/sRGB.icc result.png`

![result2](https://user-images.githubusercontent.com/863699/93179990-c1351e80-f736-11ea-9447-c55b2a443f1b.png)

**Comment by snibgo** [comment]
One of your commands merges a CMYK against a sRGB canvas, which is guaranteed to create problems.

When you have a CMYK with embedded profile, and you want sRGB, I suggest you convert it to sRGB _before_ processing such as merging against a background.

```
magick bucket.tif[2] -profile sRGB.icc -size 150x106 canvas:none +swap -background none -layers merge info:
```

**Comment by fmw42** [comment]
This seems to work for me for this image. Layer 1 is transparent and the correct size.

`magick -quiet bucket.tif[1,2] -profile /Users/fred/images/profiles/sRGB.icc -background none -layers merge x.png`

![x](https://user-images.githubusercontent.com/13458218/93238821-cb571d00-f736-11ea-8a15-4542051f7ca4.png)
